File: 프로그래머스/2/42586. 기능개발/기능개발.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<int> queue = new Queue<int>();
        List<int> answerList = new List<int>();

        for (int i = 0; i < progresses.Length; i++)
        {
            // 며칠이 걸리는지 계산
            int day = 100 - progresses[i];
            int remainder = day % speeds[i];
            day /= speeds[i];

            if(remainder != 0)
            {
                day++;
            }

            queue.Enqueue(day);
        }

        while (queue.Count != 0)
        {
            int currentValue = queue.Dequeue();
            int point = 1;

            while(queue.Count != 0 && currentValue >= queue.Peek())
            {
                point++;
                queue.Dequeue();
            }

            answerList.Add(point);
        }

        return answerList.ToArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <iostream>
#include <string>
#include <queue>
#include <unordered_map>
#include "algorithm"
using namespace std;


vector<int> solution(vector<int> progresses, vector<int> speeds) {
        vector<int> answer;
    queue<int> q;

    for (size_t i = 0; i < progresses.size(); i++)
    {
        // 100 에서 뺸 값에서 speeds를 나누고 나머지가 있다면 +1
        int count = 0;
        count = (100 - progresses[i]) / speeds[i];
        if ((100 - progresses[i]) % speeds[i])
            ++count;

        q.push(count);
    }

    int count{ 0 };
    // 하나 뽑고 그 이후꺼 
    int temp = q.front();
    q.pop();
    ++count;

    while (!q.empty())
    {
        // 현재 일차가 다음 일보다 크면 올려주고 뽑아
        if (temp >= q.front())
        {
            ++count;
        }
        else    // 갈아껴야돼
        {
            answer.push_back(count);

            count = 1;

            temp = q.front();
           
        }
        q.pop();
    }

    answer.push_back(count);

    return answer;
}